udata/core/badges/models.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import logging
+from datetime import datetime
+
+from udata.models import db
+
+log = logging.getLogger(__name__)
+
+__all__ = ('Badge',)
+
+
+class Badge(db.EmbeddedDocument):
+    kind = db.StringField(choices=[], required=True)
+    created = db.DateTimeField(default=datetime.now, required=True)
+    created_by = db.ReferenceField('User')
+    removed = db.DateTimeField()
+    removed_by = db.ReferenceField('User')
+
+    meta = {
+        'allow_inheritance': True,
+        'ordering': ['created'],
```

Comment:
Given the display on the page, I think it's relevant to keep badges ordered by creation date. This way there will be "classic" ones first (public service, certified) and then additional and maybe more pertinent ones easier to scan for the visitor. I'm not sure about that though so maybe we should try and see once we've got more data?